src/algebra/pointwise.lean

```diff
@@ -893,6 +918,26 @@ def image_hom [monoid α] [monoid β] (f : α →* β) : set_semiring α →+* s
 
 end monoid
 
+section comm_monoid
+
+variable [comm_monoid α]
+
+instance : canonically_ordered_comm_semiring (set_semiring α) :=
+{ add_le_add_left := λ a b ab c, add_le_add_left ab _,
+  le_iff_exists_add := λ a b,
+    ⟨λ ab, ⟨b, le_antisymm (subset_union_right _ _)
+      (λ x xab, by { cases (mem_union _ _ _).mp xab with xa xb,
+      { exact mem_of_mem_of_subset xa ab },
+      { exact xb }})⟩,
```

Comment:
```suggestion
      (λ x, or.imp_right ab)⟩,
```

Reply:
Here also I could not get this to work, even within `classical`.  I also had to use a type annotation: is this what you had in mind?

Reply:
See below 😉